crates/rbuilder/src/blocklist/mod.rs

```diff
@@ -0,0 +1,148 @@
+use ahash::{HashSet, HashSetExt};
+use alloy_primitives::Address;
+use serde::{Deserialize, Deserializer};
+use std::fs::read_to_string;
+use std::path::PathBuf;
+
+#[allow(clippy::len_without_is_empty)]
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+pub struct BlockList {
+    list: HashSet<Address>,
+}
+
+impl BlockList {
+    fn new() -> Self {
+        Self {
+            list: HashSet::new(),
+        }
+    }
+
+    fn from_file(path: PathBuf) -> eyre::Result<Self> {
+        let blocklist_file = read_to_string(path)?;
+        let blocklist: Vec<Address> = serde_json::from_str(&blocklist_file)?;
+
+        Ok(Self {
+            list: blocklist.into_iter().collect(),
+        })
+    }
+
+    pub fn len(&self) -> usize {
+        self.list.len()
+    }
+
+    pub fn contains(&self, address: &Address) -> bool {
+        self.list.contains(address)
+    }
```

Comment:
Might want to impl

```rust
impl Deref for BlockList {
    type Target = HashSet<Address>;

    fn deref(&self) -> &Self::Target {
        &self.list
    }
}
```

instead of the inner methods individually

Reply:
Added Deref but not sure what changes I have to make to the len and contains functions.

Reply:
You can remove them now :)